src/game_events/manager_impl.hpp

```diff
@@ -20,8 +20,26 @@
 #include <deque>
 #include <unordered_map>
 
+class game_lua_kernel;
+
 namespace game_events
 {
+class event_handlers;
+class pending_event_handler
+{
+	event_handlers& list_;
+	handler_ptr handler_;
+public:
+	pending_event_handler(event_handlers& list, handler_ptr handler)
+		: list_(list)
+		, handler_(handler)
+	{}
+	bool valid() const {return handler_.get();}
+	event_handler* operator->() {return handler_.get();}
+	event_handler& operator*() {return *handler_;}
+	~pending_event_handler();
```

Comment:
Rule of Three, although for this one I think it should be moveable but not copyable.